rmf_site_format/src/misc.rs

```diff
@@ -441,10 +441,44 @@ pub struct PreviewableMarker;
 
 /// This component is applied to each site element that gets loaded in order to
 /// remember what its original ID within the Site file was.
-#[derive(Clone, Copy, Debug)]
+#[derive(Serialize, Deserialize, Clone, Copy, Debug)]
 #[cfg_attr(feature = "bevy", derive(Component, Deref, DerefMut))]
 pub struct SiteID(pub u32);
 
+/// This component is applied to an entity as a reference to its parent entity.
+#[derive(Serialize, Deserialize, Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+#[serde(transparent)]
+#[cfg_attr(feature = "bevy", derive(Component, Reflect))]
+pub struct SiteParent<T: RefTrait>(pub Option<T>);
```

Comment:
I might have found a case for having a `SiteParent` component, unsure if it was the original intention: when saving model instances to file, we need to export their parent entities _if_ they exist (i.e. which level the model instance is spawned on). This data is stored as an `Option`, which is not a valid component for the `ModelInstance` bundle. This is where having a `SiteParent` wrapping the `Option` could be useful.

Additionally, the `SiteParent` was never used when loading instances. When spawning instance entities, the parents are always set to the `Site` entity in `load.rs` first, but later updated by checking the `SiteParent` data in [this system](https://github.com/open-rmf/rmf_site/blob/xiyu/scenarios/rmf_site_editor/src/site/model.rs#L816). I have since modified it to use the imported `SiteParent` directly during loading in [a850590](https://github.com/open-rmf/rmf_site/pull/242/commits/a8505907321595432a0cf4bb849024d79bcf138a).

The follow-up for this could be to do away with the `Option` and make it mandatory for all instances to have a `Parent`? Doing away with orphan instances in files.

Reply:
I like the sound of making it mandatory for all instances to have a `Parent`. The instance spawning workflow should generally be able to ensure a proper parent when an instance is created. In the strange case of an instance that doesn't have a `Parent`, the save function should just ignore it. Maybe we can also add a diagnostic for that case.